more volumes shorthand

ublubu opened the pull request:
#94 #77 

builds on #106 

@wlan0 

Still in progress: config-map, secret, downward-api, git, projected, persistent volumes

wlan0 commented on testdata/pods/pod_spec_with_volume_source_aws_ebs.short.yaml:
```diff
@@ -8,9 +8,9 @@ pod:
+    aws_ebs_test_volume:
+      fs: xfs
+      ro: true
+      vol_name: i4054053
```
This should be called ID

wlan0 commented on testdata/pods/pod_spec_with_volume_source_azure_disk.short.yaml:
```diff
@@ -8,12 +8,11 @@ pod:
+      disk_uri: disk://uri.azure.disk
+      fs: xfs
+      kind: dedicated
+      vol_name: azure_disk_name
```
disk_name: $disk_name

`vol_name` is confusing.

wlan0 commented on testdata/pods/pod_spec_with_volume_source_cinder.short.yaml:
```diff
@@ -8,9 +8,9 @@ pod:
+    test_volume:
+      fs: xfs
+      ro: true
+      vol_name: cinderID
```
vol_id instead of name everywhere?

ublubu replied:
sgtm

ublubu commented:
updated, @wlan0 
switched vol_name to vol_id
for azure_disk, moved vol_id to disk_name
for glusterfs, moved path to vol_id, moved vol_id to endpoints. (both fields are required) according to the docs, path is the volume id, so it makes more sense this way.

wlan0 commented:
Thanks, LGTM
